### PR TITLE
Upgrade to ndc-spec v0.1.1.

### PR DESCRIPTION
**Pull request by SamirTalwar**
This includes:

- type representations in the models
- upgrading the OpenTelemetry SDK
- adding a feature switch to support `rustls` instead of `native-tls`
- a lot of enhancements to ndc-test

### CODE REVIEW

File: crates/sdk/src/default_main.rs [hunk 1]
```diff
@@ -228,8 +228,11 @@ where
     <Setup::Connector as Connector>::Configuration: Clone,
     <Setup::Connector as Connector>::State: Clone,
 {
-    init_tracing(&serve_command.service_name, &serve_command.otlp_endpoint)
-        .expect("Unable to initialize tracing");
+    init_tracing(
```
Comment (plcplc) [inline crates/sdk/src/default_main.rs:231]: What's the rationale for prefering to `.as_deref()` over `&` ?

File: crates/sdk/Cargo.toml [hunk 2]
```diff
@@ -13,37 +13,40 @@ name = "ndc_hub_example"
...
-async-trait = "^0.1.74"
-axum = "^0.6.20"
-axum-extra = "^0.8.0"
-bytes = "1.5.0"
-clap = { version = "^4.4.7", features = ["derive", "env"] }
-http = "^0.2"
+ndc-client = { git = "http://github.com/hasura/ndc-spec.git", tag = "v0.1.1" }
+ndc-test = { git = "http://github.com/hasura/ndc-spec.git", tag = "v0.1.1", optional = true }
+
+async-trait = "0.1.79"
```
Comment (plcplc) [inline crates/sdk/Cargo.toml:27]: How do these version upgrades come about? Is it just `cargo update` or somesuch?

File: crates/sdk/src/default_main.rs (hunk 1 above)
Comment (SamirTalwar) [reply crates/sdk/src/default_main.rs:231]: It transforms `Option<String>` to `Option<&str>`.

File: crates/sdk/Cargo.toml (hunk 2 above)
Comment (SamirTalwar) [reply crates/sdk/Cargo.toml:27]: `cargo upgrade`. I was upgrading `opentelemetry` and friends to match ndc-client and figured I might as well do the rest (except for upgrades to the HTTP stack, which are incompatible).

**Review by plcplc — APPROVED** [review]

**Pull request merged**